### PR TITLE
Fix `Set A Cookie` Algorithm

### PR DESCRIPTION
**Pull request by ayuishii**
Addresses issue: https://github.com/WICG/cookie-store/issues/227

* Updates "Set a Cookie" algorithm to not null check `path` as it should always be populated
* Updates  `set` method to call algorithm with all non-optional parameters

***
<a href="https://pr-preview.s3.amazonaws.com/ayuishii/cookie-store/pull/231.html" title="Last updated on Jun 7, 2024, 11:54 PM UTC (8e63f45)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/cookie-store/231/7f81399...ayuishii:8e63f45.html" title="Last updated on Jun 7, 2024, 11:54 PM UTC (8e63f45)">Diff</a>

### CODE REVIEW

**Comment by inexorabletash** [comment]
``` [INFO] Build bikeshed document "index.bs" (common)…
    $ bikeshed --die-on=fatal spec "index.bs" "index.bs.built.html" 
      LINE ~1271: Couldn't find section '#section-5.7.3' in spec 'rfc6265bis':
      [[RFC6265bis#section-5.7.3]]
       ✘  Did not generate, due to errors exceeding the allowed error level.
    Command `bikeshed --die-on=fatal spec "index.bs" "index.bs.built.html" ` failed with exit code: 2.
```

Unfortunately the RFC renumbered things. Fortunately Bikeshed actually catches this now, which is awesome. I think I made a local PR to update the numbers a while ago, which may or may not still be valid. Let me take a look...

**Review by inexorabletash — APPROVED** [review]
Changes LGTM, pending link fixes.

**Comment by inexorabletash** [comment]
Ah, the latest rfc6265bis has named anchors. Let me see if I can get a PR together quickly...

File: index.bs [hunk 1]
```diff
@@ -43,6 +43,8 @@ spec: html; urlPrefix: https://html.spec.whatwg.org/multipage/
 <pre class=link-defaults>
 spec:service-workers; type:dfn; for:/; text:service worker
 spec:service-workers; type:dfn; for:/; text:service worker registration
+spec:dom; type:dfn; text:document
```
Comment (inexorabletash) [inline index.bs:46]: These shouldn't be necessary... are you testing with an up-to-date Bikeshed?

**Comment by ayuishii** [comment]
Wohoo, passed! Thanks @inexorabletash !

**Pull request merged**